gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -0,0 +1,257 @@
+   * used to issue requests.
+   * 
+   * @param storage the storage service used for issuing requests
+   * @param bucket bucket's name
+   * @param blob blob's name
+   */
+  public Blob(Storage storage, String bucket, String blob) {
+    this.storage = checkNotNull(storage);
+    this.info = BlobInfo.of(checkNotNull(bucket), checkNotNull(blob));
+  }
+
+  /**
+   * Return the blob's information.
+   */
+  public BlobInfo info() {
+    return info;
+  }
+
+  /**
+   * Check if this blob exists.
+   *
+   * @return true if this blob exists, false otherwise
+   * @throws StorageException upon failure
+   */
+  public boolean exists() {
+    return storage.get(info.bucket(), info.name()) != null;
+  }
+
+  /**
+   * Return this blob's content.
+   *
+   * @param options blob read options
+   * @throws StorageException upon failure
+   */
+  public byte[] content(Storage.BlobSourceOption... options) {
+    return storage.readAllBytes(info.bucket(), info.name(), options);
+  }
+
+  /**
+   * Fetches current blob's latest information.
+   *
+   * @param options blob read options
+   * @return a {@code Blob} object with latest information
+   * @throws StorageException upon failure
+   */
+  public Blob reload(BlobSourceOption... options) {
+    return new Blob(storage, storage.get(info.bucket(), info.name(), convert(info, options)));
+  }
+
+  /**
+   * Update the blob's information. Bucket or blob's name cannot be changed by this method. If you
+   * want to rename the blob or move it to a different bucket use the {@link #copyTo} and
+   * {@link #delete} operations. A new {@code Blob} object is returned. By default no checks are
+   * made on the metadata generation of the current blob. If you want to update the information only
+   * if the current blob metadata are at their latest version use the {@code metagenerationMatch}
+   * option: {@code blob.update(newInfo, BlobTargetOption.metagenerationMatch()}.
+   *
+   * @param blobInfo new blob's information. Bucket and blob names must match the current ones
+   * @param options update options
+   * @return a {@code Blob} object with updated information
+   * @throws StorageException upon failure
+   */
+  public Blob update(BlobInfo blobInfo, BlobTargetOption... options) {
+    checkArgument(Objects.equals(blobInfo.bucket(), info.bucket()), "Bucket name must match");
+    checkArgument(Objects.equals(blobInfo.name(), info.name()), "Blob name must match");
+    return new Blob(storage, storage.update(blobInfo, options));
+  }
+
+  /**
+   * Delete this blob.
+   *
+   * @param options blob delete options
+   * @return true if blob was deleted
+   * @throws StorageException upon failure
+   */
+  public boolean delete(BlobSourceOption... options) {
+    return storage.delete(info.bucket(), info.name(), convert(info, options));
+  }
+
+  /**
+   * Copy this blob to the target bucket, preserving its name. Possibly update metadata.
```

Comment:
Possibly update metadata -> possibly copying also some of the metadata (e.g. content-type)?